src/Shared/FileUtilities.cs

```diff
@@ -174,6 +174,8 @@ internal static bool CanWriteToDirectory(string directory)
             try
             {
                 string testFilePath = Path.Combine(directory, $"MSBuild_{Guid.NewGuid().ToString("N")}_testFile.txt");
+                FileInfo file = new(testFilePath);
+                file.Directory.Create(); // If the directory already exists, this method does nothing.
```

Comment:
Does this work if multiple levels of the nesting don't exist? I would assume so but the docs don't explicitly say. https://docs.microsoft.com/en-us/dotnet/api/system.io.directoryinfo.create?view=net-6.0

Reply:
Yes. I tried `newFolder\otherFolder\thirdFolder` from a folder that existed (where none of those folders existed previously), and they were all made.